### DISCUSSION TITLE
NextJS and headless WordPress performance issues

### DISCUSSION THREAD

**Discussion by dimitryvolos** [wp-graphql/wp-graphql#2820] [Q&A]
Hi,

I am building a Nextjs + Headless WordPress site.

It fetches data (posts and page data) from a server located in the UK, via WPGraphQL.

In development mode and only local machine it is working fast and without issues (still fetching data from a remote UK-based server).

But once deployed to Netlify, a small 5-post GraphQL request will take a few seconds to return a response. Larger queries will overload the server.

Could this be the problem with server locations for Netlify build?

I am also using stellate.co free tear, which does seem to help a little.

How come the local dev version fetches data in seconds, and production is so much slower?

Any help is much appreciated.

Regards,
Dimitry

**Comment by kau-nakrani** [comment]
Hi @dimitryvolos 

Will there be a demand for wordpress + next.js (or any js) after 2026?

Thank you
